doc/cifarSC2011/advanced_theano.txt

```diff
@@ -179,96 +180,189 @@ Inplace optimization
-        0.4%   98.7%  0.127s  29.021s  1.27e-05s   10000  1 Alloc
-        0.3%   99.0%  0.092s  29.112s  9.16e-06s * 10000  1 Elemwise{Composite{exp,{mul,{true_div,neg,{add,mul}}}}}[(0, 0)]
-        0.1%   99.3%  0.033s  29.265s  1.66e-06s * 20001  3 InplaceDimShuffle{x}
-       ... (remaining 11 Apply account for 0.7%(0.00s) of the runtime)
-    (*) Op is running a c implementation
-
-    Apply-wise summary:
-    <% of local_time spent at this position> <cumulative %%>
-        <apply time> <cumulative seconds> <time per call>
-        <nb_call> <Apply position> <Apply Op name>
-       87.3%   87.3%  25.672s  25.672s 2.57e-03s  10000  15 Gemv{inplace}(w, TensorConstant{-0.01}, InplaceDimShuffle{1,0}.0, Elemwise{Composite{exp,{mul,{true_div,neg,{add,mul}}}}}[(0, 0)].0, TensorConstant{0.9998})
-        9.7%   97.0%  2.843s  28.515s 2.84e-04s  10000   1 dot(x, w)
-        1.3%   98.2%  0.378s  28.893s 3.78e-05s  10000   9 Elemwise{Composite{scalar_softplus,{mul,scalar_softplus,{neg,mul,sub}}}}(y, Elemwise{Composite{neg,sub}}[(0, 0)].0, Elemwise{sub,no_inplace}.0, Elemwise{neg,no_inplace}.0)
-        0.4%   98.7%  0.127s  29.020s 1.27e-05s  10000  10 Alloc(Elemwise{inv,no_inplace}.0, Shape_i{0}.0)
-        0.3%   99.0%  0.092s  29.112s 9.16e-06s  10000  13 Elemwise{Composite{exp,{mul,{true_div,neg,{add,mul}}}}}[(0,0)](Elemwise{ScalarSigmoid{output_types_preference=transfer_type{0}, _op_use_c_code=True}}[(0, 0)].0, Alloc.0, y, Elemwise{Composite{neg,sub}}[(0,0)].0, Elemwise{sub,no_inplace}.0, InplaceDimShuffle{x}.0)
-        0.3%   99.3%  0.080s  29.192s 7.99e-06s  10000  11 Elemwise{ScalarSigmoid{output_types_preference=transfer_type{0}, _op_use_c_code=True}}[(0, 0)](Elemwise{neg,no_inplace}.0)
-       ... (remaining 14 Apply instances account for
-           0.7%(0.00s) of the runtime)
-
-    Profile of Theano functions memory:
-    (This check only the output of each apply node. It don't check the temporary memory used by the op in the apply node.)
-    Theano fct: train
-        Max without gc, inplace and view (KB) 2481
-        Max FAST_RUN_NO_GC (KB) 16
-        Max FAST_RUN (KB) 16
-        Memory saved by view (KB) 2450
-        Memory saved by inplace (KB) 15
-        Memory saved by GC (KB) 0
-        <Sum apply outputs (bytes)> <Apply outputs memory size(bytes)>
-            <created/inplace/view> <Apply node>
-        <created/inplace/view> is taked from the op declaration, not ...
-             2508800B  [2508800] v InplaceDimShuffle{1,0}(x)
-                6272B  [6272] i Gemv{inplace}(w, ...)
-                3200B  [3200] c Elemwise{Composite{...}}(y, ...)
-
-    Here are tips to potentially make your code run faster (if you think of new ones, suggest them on the mailing list).
-    Test them first, as they are not guaranteed to always provide a speedup.
-      - Try the Theano flag floatX=float32
+    Function profiling
```

Comment:
That's what Mehdi told me. It seems that now the data for the memory profile is all weird (it says 0KB memory usage everywhere) but that would be a bug with the memory profile rather than with this PR. What do you think, @lamblin ?